core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/CloneWorkspaceContainerRunnerSpec.scala

```diff
@@ -0,0 +1,219 @@
+package org.broadinstitute.dsde.rawls.monitor.workspace.runners
+
+import bio.terra.workspace.model.JobReport
+import org.broadinstitute.dsde.rawls.TestExecutionContext
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, SamDAO, SlickDataSource}
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkspaceManagerResourceMonitorRecord
+import org.broadinstitute.dsde.rawls.dataaccess.workspacemanager.WorkspaceManagerDAO
+import org.broadinstitute.dsde.rawls.model.{RawlsUserEmail, Workspace}
+import org.broadinstitute.dsde.rawls.monitor.workspace.runners.CloneWorkspaceContainerRunnerSpec.{
+  monitorRecord,
+  userEmail,
+  workspace,
+  workspaceId
+}
+import org.joda.time.DateTime
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito.{doAnswer, doReturn, spy, verify}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.mockito.MockitoSugar
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+
+object CloneWorkspaceContainerRunnerSpec {
+  val userEmail: String = "user@email.com"
+  val workspaceId: UUID = UUID.randomUUID()
+  val wsCreatedDate: DateTime = DateTime.parse("2023-01-18T10:08:48.541-05:00")
+  val monitorRecord: WorkspaceManagerResourceMonitorRecord =
+    WorkspaceManagerResourceMonitorRecord.forCloneWorkspaceContainer(
+      UUID.randomUUID(),
+      workspaceId,
+      RawlsUserEmail(userEmail)
+    )
+
+  val workspace: Workspace = Workspace(
+    "test-ws-namespace",
+    "test-ws-name",
+    workspaceId.toString,
+    "test-bucket",
+    None,
+    wsCreatedDate,
+    wsCreatedDate,
+    "a_user",
+    Map()
+  )
+}
+
+class CloneWorkspaceContainerRunnerSpec extends AnyFlatSpecLike with MockitoSugar with Matchers with ScalaFutures {
+  implicit val executionContext: ExecutionContext = TestExecutionContext.testExecutionContext
+
+  behavior of "initial setup and basic requirements updating workspace container cloning status monitoring"
+
+  it should "return a completed status if the workspace id is not set on the job" in {
+    val runner = new CloneWorkspaceContainerRunner(
+      mock[SamDAO],
+      mock[WorkspaceManagerDAO],
+      mock[SlickDataSource],
+      mock[GoogleServicesDAO]
+    )
+    whenReady(runner(monitorRecord.copy(workspaceId = None)))(
+      _ shouldBe WorkspaceManagerResourceMonitorRecord.Complete
+    )
+  }
+
+  it should "return a completed status if no workspace is found for the id" in {
```

Comment:
Not sure I really understand this test case. 

Is this the case where we clone a workspace and the destination workspace isn't found in Rawls after it's finished cloning in WSM? That seems like it should raise an error or otherwise fail in some way

Reply:
This is just that it completes the _job_ if the workspace is not there: `Complete` vs `Incomplete` is just whether or not the runner is finished with the job.
In this case, an error is logged, but we can't set the message on the workspace, because... well, no workspace.

In practice, this is an unlikely scenario, but it could occur with some manual intervention